File: Aula_05/python/01.py
import math

class Circulo:
    def __init__ (self):
        self.__raio = 0

    def set_raio (self, v):
        if v > 0:
            self.__raio = v
        else:
            raise ValueError ("A base do triangulo nao pode ser negativa")
        
    def get_raio (self):
        return self.__raio
    
    def calc_area (self):
        return self.__raio ** 2 * math.pi
    
    def calc_circunferencia (self):
        return 2 * math.pi * self.__raio
    

class UI:
    @staticmethod
    def main():
        x = Circulo()

        x.set_raio(float(input("Informe o valor do raio do circulo: ")))

        print(f"Raio = {x.get_raio()}")
        print(f"Area = {x.calc_area():.2f}")
        print(f"Circunferencia = {x.calc_circunferencia():.2f}")
        
UI.main()

File: Aula_05/c++/01.cpp
#include <iostream>
#include <math.h>

class Circulo{
    private:
    double raio = 0;

    public: 
    void setRaio (double v) {
        if (v > 0){
            raio = v;
        } else {
            throw std::invalid_argument ("o raio do circulo nao pode ser um valor negativo");
        }
    }

    double getRaio () {
        return raio;
    }

    double calcArea () {
        return std::pow(raio, 2) * M_PI;
    }

    double calcCircunferencia () {
        return 2 * M_PI * raio;
    }
};

class UI{
    public:
    static void main(){
        Circulo x;

        double valor;
        std::cout << "Informe o valor do raio do circulo: ";
        std::cin >> valor;
        x.setRaio(valor);

        std::cout << "Raio = " << x.getRaio() << std::endl;
        std::cout << "Area = " << x.calcArea() << std::endl;
        std::cout << "Circunferencia = " << x.calcCircunferencia() << std::endl;
    }
};

int main(){
    UI::main();
    return 0;
}